frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -114,7 +114,7 @@ export function getDashboardUiParameters(
   metadata: Metadata,
 ): UiParameter[] {
   const { parameters, dashcards } = dashboard;
-  const mappings = getMappings(dashcards as DashboardCard[]);
+  const mappings = getMappings(dashcards as QuestionDashboardCard[]);
```

Comment:
We can filter out `dashcards` using a type-guard to avoid the cast.
Same goes for `hasMatchingParameters` below.

Reply:
Fixed in [`bcd3dc2`](https://github.com/metabase/metabase/pull/38444/commits/bcd3dc2cceff4ae7fe5b84414554d1b02964db51)